### PR TITLE
Add WireGuard plugin

### PR DESCRIPTION
**Pull request by PaulosV**
Adding new plugin for detecting WireGuard traffic.

It parses three basic fields: packet type, ephemeral identifiers of sender and receiver. It also creates new flow for each handshake initiation.

### CODE REVIEW

File: wgplugin.cpp [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+   }
+
+   ext->possible_wg = 100;
+   identified++;
+   return true;
+}
+
+int WGPlugin::add_ext_wg(const char *data, unsigned int payload_len, bool source_pkt, Flow &rec)
+{
+   RecordExtWG *ext = new RecordExtWG();
```
Comment (havraji6) [inline wgplugin.cpp:221]: There is a memory allocation each time UDP packet arrives. Better solution is to preallocate record.

File: Makefile.am [hunk 2]
```diff
@@ -82,7 +82,9 @@ ipfixprobe_src=main.cpp \
 		ipfix-basiclist.cpp \
 		ipfix-basiclist.h \
 		basicplusplugin.h \
-		basicplusplugin.cpp
+		basicplusplugin.cpp \
+		wgplugin.h \
+		wgplugin.cpp
```
Comment (havraji6) [inline Makefile.am:89]: Missing `traffic-samples/wg-sample.pcap` in `EXTRA_DIST` variable.

File: wgplugin.cpp [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+   uint8_t pkt_type = data[0];
+   if (pkt_type < WG_PACKETTYPE_INIT_TO_RESP || pkt_type > WG_PACKETTYPE_TRANSPORT_DATA) {
+      return false;
+   }
+   if (data[1] != 0x0 || data[2] != 0x0 || data[3] != 0x0) {
+      return false;
+   }
+
+   // Next, check the packet contents based on the message type.
+   switch(pkt_type) {
```
Comment (havraji6) [inline wgplugin.cpp:163]: Fix coding style in this method.
`switch(` -> `switch (`
`(data+4)` -> `(data + 4)`
`(! source_pkt)` -> `(!source_pkt)`

**Review by havraji6 — CHANGES_REQUESTED** [review]
Don't forget to update `README.md`.

File: Makefile.am (hunk 2 above)
Comment (PaulosV) [reply Makefile.am:89]: Ah, good catch, will fix.

File: wgplugin.cpp (hunk 1 above)
Comment (PaulosV) [reply wgplugin.cpp]: OK, got it now, should be fixed.

**Review by havraji6 — APPROVED** [review]

**Pull request merged**